### PR TITLE
RFC 5256 - SORT command

### PR DESCRIPTION
**Pull request by soywod**
#117

### CODE REVIEW

File: imap-proto/src/parser/rfc5256.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    sequence::{preceded, terminated},
+    IResult,
+};
+
+use crate::{parser::core::number, types::MailboxDatum};
+
+pub(crate) fn mailbox_data_sort(i: &[u8]) -> IResult<&[u8], MailboxDatum> {
+    map(
+        // Technically, trailing whitespace is not allowed here, but multiple
+        // email servers in the wild seem to have it anyway (see #34, #108).
```
Comment (djc) [inline imap-proto/src/parser/rfc5256.rs]: Have you actually seen this for this particular extension?

File: imap-proto/src/parser/rfc5256.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    bytes::streaming::{tag, tag_no_case},
+    combinator::{map, opt},
+    multi::many0,
+    sequence::{preceded, terminated},
+    IResult,
+};
+
+use crate::{parser::core::number, types::MailboxDatum};
+
+pub(crate) fn mailbox_data_sort(i: &[u8]) -> IResult<&[u8], MailboxDatum> {
```
Comment (djc) [inline imap-proto/src/parser/rfc5256.rs:34]: Can you add a docstring here with the syntax from the relevant RFC, and maybe a little context on what this is/how it's used? (There are many existing parsers that have stuff like this, see #116 for some examples.)

File: imap-proto/src/parser/rfc5256.rs (hunk 1 above)
Comment (soywod) [reply imap-proto/src/parser/rfc5256.rs]: In fact the `SORT` command is very close to the `SEARCH` command (it kind of extends it). So I guess they share the same "issues". Should I remove?

Comment (djc) [reply imap-proto/src/parser/rfc5256.rs]: I guess it's fine to keep, but may make it explicit that this is conjecture based on the similarity to `SEARCH`?

**Comment by djc** [comment]
Thanks!

**Comment by soywod** [comment]
Thank YOU for this great tool!

**Comment by jonhoo** [comment]
@djc Ah, this change was not backwards-compatible, since `MailboxDatum` isn't declared as `#[non_exhaustive]` (even though it probably should be). May be worth adding that attribute and going `0.14`?

**Comment by djc** [comment]
Ah, sorry about that! Will do

**Comment by djc** [comment]
Published 0.14.0 and yanked 0.13.1.

**Pull request merged**